cmd/api-server/main.go

```diff
@@ -331,3 +332,38 @@ func main() {
 		log.DefaultLogger.Fatalf("Testkube is shutting down: %v", err)
 	}
 }
+
+// getMongoSSLConfig builds the necessary SSL connection info from the settings in the environment variables
+// and the given secret reference
+func getMongoSSLConfig(c MongoConfig, secretClient *secret.Client) *storage.MongoSSLConfig {
+	if c.SSLSecretRef == "" {
+		return nil
+	}
+
+	mongoSSLSecret, err := secretClient.Get(Config.SSLSecretRef)
+	ui.ExitOnError(fmt.Sprintf("Could not get secret %s for MongoDB connection", c.SSLSecretRef), err)
+
+	var keyFile, caFile, pass string
+	var ok bool
+	if keyFile, ok = mongoSSLSecret["sslClientCertificateKeyFile"]; !ok {
+		ui.Warn("Could not find sslClientCertificateKeyFile in secret %s", c.SSLSecretRef)
+	}
+	if caFile, ok = mongoSSLSecret["sslCertificateAuthorityFile"]; !ok {
+		ui.Warn("Could not find sslCertificateAuthorityFile in secret %s", c.SSLSecretRef)
+	}
+	if pass, ok = mongoSSLSecret["sslClientCertificateKeyFilePassword"]; !ok {
+		ui.Warn("Could not find sslClientCertificateKeyFilePassword in secret %s", c.SSLSecretRef)
+	}
+
+	err = os.WriteFile("mongodb.pem", []byte(keyFile), 0644)
```

Comment:
Should we place in a server current dir or might be in some temp folder?

Reply:
good point, will update it